controllers/targetsecret_controller_test.go

```diff
@@ -140,15 +140,36 @@ var _ = Describe("Samlet Controller", func() {
 		It("Expire time is properly calculated", func() {
 			deadline := &metav1.Time{Time: time.Now().Add(time.Duration(9) * time.Minute)}
 			By("Delete existing saml2aws")
-			Expect(k8sClient.Delete(context.Background(), obj)).Should(Succeed())
+			Eventually(func() error {
+				return k8sClient.Delete(context.Background(), obj)
+			}, timeout, interval).Should(Succeed())
 			By("Specifying session duration 30 minutes")
 			obj.Spec.SessionDuration = "30m"
 			obj.ObjectMeta.ResourceVersion = ""
 			By("Creating test saml object")
-			Expect(k8sClient.Create(context.Background(), obj)).Should(Succeed())
+			Eventually(func() error {
+				return k8sClient.Create(context.Background(), obj)
+			}, timeout, interval).Should(Succeed())
 			By("Checking expire time")
 			Expect(obj.Status.ExpirationTime.Before(deadline)).Should(BeTrue())
 
 		})
+		It("sessionDuration overrides global SESSION_DURATION option", func() {
```

Comment:
When we decided to use BDD framework, we should stick to its DSL semantics in order to express desired system behavior, otherwise BDD framework doesn't bring any value comparing to classic TDD framework.
It() closure describes behavior of functionality in Describe(), so it should read naturally, e.g.:
`It("prefers sessionDuration from spec over global SESSION_DURATION option", func() {`
instead of 
`It("sessionDuration overrides global SESSION_DURATION option", func() {`

Reply:
fixed, thanks